src/src/PluginStructs/P104_data_struct.h

```diff
@@ -0,0 +1,440 @@
+# endif  // ifdef P104_MINIMAL_ANIMATIONS
+
+# ifdef P104_MEDIUM_ANIMATIONS
+#  define ENA_SPRITE  0 // Disabling more MD_Parola features
+#  define ENA_OPNCLS  0
+#  define ENA_SCR_DIA 0
+#  define ENA_GROW    0
+# endif // ifdef P104_MEDIUM_ANIMATIONS
+
+# include <MD_Parola.h>
+# include <MD_MAX72xx.h>
+
+// WARNING: Order of values should match the numeric order of P104_OFFSET_* values
+# define P104_OFFSET_SIZE         0u
+# define P104_OFFSET_TEXT         1u
+# define P104_OFFSET_CONTENT      2u
+# define P104_OFFSET_ALIGNMENT    3u
+# define P104_OFFSET_ANIM_IN      4u
+# define P104_OFFSET_SPEED        5u
+# define P104_OFFSET_ANIM_OUT     6u
+# define P104_OFFSET_PAUSE        7u
+# define P104_OFFSET_FONT         8u
+# define P104_OFFSET_LAYOUT       9u
+# define P104_OFFSET_SPEC_EFFECT  10u
+# define P104_OFFSET_OFFSET       11u
+# define P104_OFFSET_BRIGHTNESS   12u
+# define P104_OFFSET_REPEATDELAY  13u
+# define P104_OFFSET_ACTION       14u
+
+# define P104_OFFSET_COUNT        15u // Highest P104_OFFSET_* defined + 1
+
+# define P104_CONFIG_ZONE_COUNT   PCONFIG(0)
+# define P104_CONFIG_TOTAL_UNITS  PCONFIG(1)
+# define P104_CONFIG_HARDWARETYPE PCONFIG(2)
+# define P104_CONFIG_FLAGS        PCONFIG_LONG(0)
+# define P104_CONFIG_DATETIME     PCONFIG_LONG(1)
+
+# define P104_CONFIG_FLAG_CLEAR_DISABLE 0
+# define P104_CONFIG_FLAG_LOG_ALL_TEXT  1
+# define P104_CONFIG_FLAG_ZONE_ORDER    2
+
+# define P104_CONFIG_DATETIME_FLASH     0
+# define P104_CONFIG_DATETIME_12H       1
+# define P104_CONFIG_DATETIME_AMPM      2
+# define P104_CONFIG_DATETIME_YEAR4DGT  3
+# define P104_CONFIG_DATETIME_FORMAT    4 // Uses 4 bits, leave some space for a few options
+# define P104_CONFIG_DATETIME_SEP_CHAR  8 // Uses 4 bits
+
+# define P104_DATE_FORMAT_EU        0
+# define P104_DATE_FORMAT_US        1
+# define P104_DATE_FORMAT_JP        2
+
+# define P104_DATE_SEPARATOR_SPACE  0
+# define P104_DATE_SEPARATOR_SLASH  1
+# define P104_DATE_SEPARATOR_DASH   2
+# define P104_DATE_SEPARATOR_DOT    3
+
+# define P104_ACTION_NONE         0
+# define P104_ACTION_ADD_ABOVE    1
+# define P104_ACTION_ADD_BELOW    2
+# define P104_ACTION_DELETE       3
+
+# define P104_CONTENT_TEXT        0
+# define P104_CONTENT_TEXT_REV    1
+# define P104_CONTENT_TIME        2
+# define P104_CONTENT_TIME_SEC    3
+# define P104_CONTENT_DATE4       4
+# define P104_CONTENT_DATE6       5
+# define P104_CONTENT_DATE_TIME   6
+# define P104_CONTENT_BAR_GRAPH   7
+# ifdef P104_USE_BAR_GRAPH
+#  define P104_CONTENT_count       8 // The number of content type options
+# else // ifdef P104_USE_BAR_GRAPH
+#  define P104_CONTENT_count       7 // The number of content type options
+# endif // ifdef P104_USE_BAR_GRAPH
+
+# define P104_SPECIAL_EFFECT_NONE       0
+# define P104_SPECIAL_EFFECT_UP_DOWN    1
+# define P104_SPECIAL_EFFECT_LEFT_RIGHT 2
+# define P104_SPECIAL_EFFECT_BOTH       P104_SPECIAL_EFFECT_UP_DOWN + P104_SPECIAL_EFFECT_LEFT_RIGHT // Used as a bitmap
```

Comment:
Wrap this define in (), or else it will be very tricky to use.

```c++
# define P104_SPECIAL_EFFECT_BOTH       (P104_SPECIAL_EFFECT_UP_DOWN + P104_SPECIAL_EFFECT_LEFT_RIGHT) // Used as a bitmap